scripts/setup-adapters.sh

```diff
@@ -23,6 +23,111 @@ SCRIPTDIR=$(dirname "${BASH_SOURCE[0]}")
 source $SCRIPTDIR/setup-helper-functions.sh
 DEPENDENCY_DIR=${DEPENDENCY_DIR:-$(pwd)}
 
+function install_gcs-sdk-cpp {
+  user=`whoami`
+  case "$user" in
+    root) echo 'Running as root' ; sudocmd="";;
+       *) echo 'Running as other user than root' ; sudocmd="sudo";;
+  esac
+  # install gcs dependencies
+  # https://github.com/googleapis/google-cloud-cpp/blob/main/doc/packaging.md#required-libraries
+  # install abseil
+  echo " Installing abseil..."
+  mkdir -p ${DEPENDENCY_DIR}/abseil-cpp && cd ${DEPENDENCY_DIR}/abseil-cpp
+  curl -sSL https://github.com/abseil/abseil-cpp/archive/20211102.0.tar.gz | \
+      tar -xzf - --strip-components=1 && \
+      sed -i 's/^#define ABSL_OPTION_USE_\(.*\) 2/#define ABSL_OPTION_USE_\1 0/' "absl/base/options.h" && \
+      cmake \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DBUILD_TESTING=OFF \
+        -DBUILD_SHARED_LIBS=yes \
+        -DCMAKE_CXX_STANDARD=11 \
+        -H. -Bcmake-out && \
+      cmake --build cmake-out -- -j ${NCPU:-4} && \
+  $sudocmd cmake --build cmake-out --target install -- -j ${NCPU:-4} && \
+  $sudocmd ldconfig
```

Comment:
For each and every of below steps it would go then:
```suggestion
      sed -i 's/^#define ABSL_OPTION_USE_\(.*\) 2/#define ABSL_OPTION_USE_\1 0/' "absl/base/options.h"
      cmake_install  -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=yes -DCMAKE_INSTALL_PREFIX="${DEPENDENCY_DIR}/install" -DCMAKE_INSTALL_PREFIX="${DEPENDENCY_DIR}/install"
      ldconfig && \
      cd ..
```